File: entities/otp.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type OTPPurpose string

const (
	OTPPurposeRegister       OTPPurpose = "REGISTER"
	OTPPurposeLogin          OTPPurpose = "LOGIN"
	OTPPurposePasswordReset OTPPurpose = "PASSWORD_RESET"
)

type OTPVerification struct {
	ID           uuid.UUID  `gorm:"primaryKey" json:"id"`
	DoctorID     uuid.UUID  `gorm:"not null;index" json:"doctor_id"`
	Purpose      OTPPurpose `gorm:"size:50;not null" json:"purpose"`
	OTP_Hash     string     `gorm:"size:255;not null" json:"-"`
	ExpiresAt    time.Time  `gorm:"not null;index" json:"expires_at"`
	IsVerified   bool       `gorm:"default:false;not null" json:"is_verified"`
	AttemptCount int        `gorm:"default:0;not null" json:"attempt_count"`
	MaxAttempts  int        `gorm:"default:5;not null" json:"max_attempts"`
	CreatedAt    time.Time  `gorm:"autoCreateTime" json:"created_at"`
}

func (o *OTPVerification) BeforeCreate(tx *gorm.DB) (err error) {
	if o.ID == uuid.Nil {
		o.ID = uuid.New()
	}
	return nil
}

File: python_backend/entities/otp.py
import enum
import uuid
from sqlalchemy import Column, String, Integer, Boolean, DateTime, Enum, ForeignKey
from sqlalchemy.sql import func
from database import Base

class OTPPurpose(str, enum.Enum):
    REGISTER = "REGISTER"
    LOGIN = "LOGIN"
    PASSWORD_RESET = "PASSWORD_RESET"

class OTPVerification(Base):
    __tablename__ = "otp_verifications"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    purpose = Column(Enum(OTPPurpose, name="otp_purpose_enum", native_enum=False), nullable=False)
    otp_hash = Column(String(255), nullable=False)
    expires_at = Column(DateTime(timezone=True), nullable=False, index=True)
    is_verified = Column(Boolean, default=False, nullable=False)
    attempt_count = Column(Integer, default=0, nullable=False)
    max_attempts = Column(Integer, default=5, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
